### DISCUSSION TITLE
Best practices for extending to include advertisements?

### DISCUSSION THREAD

**Discussion by SelfhostedPro** [facebook/docusaurus#4324] [Q&A]
I'm switching my documentation for https://yacht.sh over to docusaurus from mkdocs. In mkdocs I was able to extend their components in order to include an advertisement but I'm not seeing documentation on extending components to include custom content. 

I just need to add a script tag for https://ethical-ad-client.readthedocs.io/en/latest/

Is there a recommended way to accomplish this?

**Comment by slorber** [comment]
I don't understand what you mean exactly, but you can "override" any Docusaurus theme component by using the swizzle cli, and you can also add script tags.

You can implement your whole theme if you need to, or override our default theme with some tweaks, or wrap an existing theme component with additional dom elements (https://v2.docusaurus.io/docs/next/using-themes#wrapping-theme-components)

**Reply by SelfhostedPro** [reply]
Ah, I think that answers my question. I haven't worked with react much (my main project is built with VueJS). I'll try this out later today and post here if I run into any trouble or with what exactly I did in case someone else is trying something similar. Thanks!
